### PR TITLE
CI,STYLE: add spell check?

### PR DESCRIPTION
**Pull request by MarcoGorelli**
IIRC it was suggested in a comment some time ago to use a spell checker, so I thought I'd make a little PR showing how `codespell` would work (for now just applied to `pandas/core`).

Is this something we'd want? False positives can be ignored in `setup.cfg` in the `[codespell]` section

### CODE REVIEW

File: pandas/core/reshape/pivot.py [hunk 1]
```diff
@@ -740,7 +740,7 @@ def _build_names_mapper(
     A row or column name is replaced if it is duplicate among the rows of the inputs,
     among the columns of the inputs or between the rows and the columns.
 
-    Paramters
+    Parameters
```
Comment (jreback) [inline pandas/core/reshape/pivot.py:743]: need to fix the underline

**Review by jreback — CHANGES_REQUESTED** [review]
comment, looks fine. i think we had this before but removed it, don't remember why.

File: pandas/core/reshape/pivot.py (hunk 1 above)
Comment (jreback) [reply pandas/core/reshape/pivot.py:743]: i guess this is not public but still let's fix

**Comment by MarcoGorelli** [comment]
> i think we had this before but removed it

I just ran `git log -G codespell` and got nothing (other than this PR)

I did however find #24299, which removed `sphinxcontrb.spelling`. That's a different tool - this one (codespell) is very fast and takes ~0.7 seconds on the whole codebase from my (non-exceptional) laptop

**Comment by jreback** [comment]
> > i think we had this before but removed it
> 
> I just ran `git log -G codespell` and got nothing (other than this PR)
> 
> I did however find (with `git log -G spell`) #24299, which removed `sphinxcontrb.spelling`. That's a different tool - this one (codespell) is very fast and takes ~0.7 seconds on the whole codebase from my (non-exceptional) laptop

ahh right that was for the docs. okie this looks fine to me (we *may* want to even clean up the exclude list at some point as well, e.g. narrow it) but that's for other isssues.

File: setup.cfg [hunk 2]
```diff
@@ -63,6 +63,9 @@ filterwarnings =
     error:The SparseArray:FutureWarning
 junit_family=xunit2
 
+[codespell]
+ignore-words-list=ba,blocs,coo,datas,fo,hist,nd,ser
```
Comment (jreback) [inline setup.cfg:67]: these here, let's open an issue and try to fix some of these 
maybe 'fo' rest might be ok

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
merge on green

**Pull request merged**